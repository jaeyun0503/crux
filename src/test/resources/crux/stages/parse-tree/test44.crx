// test nested if-else and loop statements

fn main() {
  let a: int;
  a = 0;
  let b: int;
  b = 0;
  while (a < 3) {
    if (true) {
      if (true) {
        while (b < 3) {
          printInt(5);
          b = b + 1;
        }
      } else {
        printInt(5);
      }
    }
    printInt(5);
    a = a + 1;
  }
  printInt(5);
}
