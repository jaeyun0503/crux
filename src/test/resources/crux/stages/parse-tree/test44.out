program
  declList
    decl
      functionDefn
        paramList
        stmtBlock
          stmtList
            stmt
              varDecl
                type
            stmt
              assignStmt
                designator
                expr0
                  expr1
                    expr2
                      expr3
                        literal
            stmt
              varDecl
                type
            stmt
              assignStmt
                designator
                expr0
                  expr1
                    expr2
                      expr3
                        literal
            stmt
              whileStmt
                expr0
                  expr1
                    expr2
                      expr3
                        expr0
                          expr1
                            expr2
                              expr3
                                designator
                          op0
                          expr1
                            expr2
                              expr3
                                literal
                stmtBlock
                  stmtList
                    stmt
                      ifStmt
                        expr0
                          expr1
                            expr2
                              expr3
                                expr0
                                  expr1
                                    expr2
                                      expr3
                                        literal
                        stmtBlock
                          stmtList
                            stmt
                              ifStmt
                                expr0
                                  expr1
                                    expr2
                                      expr3
                                        expr0
                                          expr1
                                            expr2
                                              expr3
                                                literal
                                stmtBlock
                                  stmtList
                                    stmt
                                      whileStmt
                                        expr0
                                          expr1
                                            expr2
                                              expr3
                                                expr0
                                                  expr1
                                                    expr2
                                                      expr3
                                                        designator
                                                  op0
                                                  expr1
                                                    expr2
                                                      expr3
                                                        literal
                                        stmtBlock
                                          stmtList
                                            stmt
                                              callStmt
                                                callExpr
                                                  exprList
                                                    expr0
                                                      expr1
                                                        expr2
                                                          expr3
                                                            literal
                                            stmt
                                              assignStmt
                                                designator
                                                expr0
                                                  expr1
                                                    expr1
                                                      expr2
                                                        expr3
                                                          designator
                                                    op1
                                                    expr2
                                                      expr3
                                                        literal
                                stmtBlock
                                  stmtList
                                    stmt
                                      callStmt
                                        callExpr
                                          exprList
                                            expr0
                                              expr1
                                                expr2
                                                  expr3
                                                    literal
                    stmt
                      callStmt
                        callExpr
                          exprList
                            expr0
                              expr1
                                expr2
                                  expr3
                                    literal
                    stmt
                      assignStmt
                        designator
                        expr0
                          expr1
                            expr1
                              expr2
                                expr3
                                  designator
                            op1
                            expr2
                              expr3
                                literal
            stmt
              callStmt
                callExpr
                  exprList
                    expr0
                      expr1
                        expr2
                          expr3
                            literal
